php/WP_CLI/Runner.php

```diff
@@ -672,6 +672,48 @@ public function before_wp_load() {
 		if ( $this->cmd_starts_with( array( 'plugin' ) ) ) {
 			$GLOBALS['pagenow'] = 'plugins.php';
 		}
+
+		$this->load_wordpress();
+
+		$this->_run_command();
+
+	}
+
+	/**
+	 * Load WordPress, if it hasn't already been loaded
+	 */
+	public function load_wordpress() {
+		static $wp_cli_is_loaded;
+		// Globals not explicitly globalized in WordPress
+		global $current_site, $current_blog, $shortcode_tags;
+
+		if ( ! empty( $wp_cli_is_loaded ) ) {
+			return;
+		}
+
+		$wp_cli_is_loaded = true;
+
+		// Load wp-config.php code, in the global scope
+		eval( $this->get_wp_config_code() );
+
+		$this->maybe_update_url_from_domain_constant();
+
+		// Load Core, mu-plugins, plugins, themes etc.
+		require WP_CLI_ROOT . '/php/wp-settings-cli.php';
+
+		// Fix memory limit. See http://core.trac.wordpress.org/ticket/14889
+		@ini_set( 'memory_limit', -1 );
+
+		// Load all the admin APIs, for convenience
+		require ABSPATH . 'wp-admin/includes/admin.php';
+
+		add_filter( 'filesystem_method', function() { return 'direct'; }, 99 );
```

Comment:
D. Crockford showed that an anonymous function should be `function ()` (with a space) to aviod confusion with a function named `function`.